modules/peers.js

```diff
@@ -736,17 +736,31 @@ Peers.prototype.list = function(options, cb) {
 	);
 };
 
+/**
+ * Gets the height of maximum number of peers at one particular height.
+ *
+ * @param {Object} options
+ * @param {string} [options.normalized=false] - Return peers in normalized (json) form
+ * @param {function} cb - Callback function
+ * @returns {setImmediateCallback} cb, err, peers
+ */
 Peers.prototype.networkHeight = function(options, cb) {
 	self.list(options, (err, peers) => {
-		if (err) {
+		if (err || peers.length === 0) {
 			return setImmediate(cb, err, 0);
 		}
-		const peersGroupedByHeight = _.groupBy(peers, 'height');
-		const popularHeights = Object.keys(peersGroupedByHeight).map(Number);
-		const networkHeight = _.max(popularHeights);
+		// count by number of peers at one height
+		const heights = _.countBy(peers, 'height');
+		const height = Object.keys(heights).reduce((curr, next) => {
```

Comment:
How about simply `const networkHeight = _(peers).countBy('height').maxBy('height').value()`?

Reply:
`countBy` returns `{5: 3, 7: 1, 10: 1}` where `5,7,10` are the heights and `3,1,1` are the corresponding counts, so I dont think this will work`const networkHeight = _(peers).countBy('height').maxBy('height').value()`

Reply:
You are right @ManuGowda. How about:
```
const mostPopularGroup = _(peers).countBy('height').map((counts, height) => {
return {
  height: +height,
  counts
};
}).maxBy('counts');

const networkHeight = mostPopularGroup.height
```